src/Enforcers/ValidationEnforcer.php

```diff
@@ -0,0 +1,148 @@
+use PDGA\DataObjects\Validators\DateValidator;
+use PDGA\DataObjects\Validators\IntValidator;
+use PDGA\DataObjects\Validators\NotNullValidator;
+use PDGA\DataObjects\Validators\StringValidator;
+use PDGA\Exception\ValidationListException;
+
+class ValidationEnforcer
+{
+    /**
+     * Validates the properties of an object against a class definition.
+     *
+     * @param array|object $object - The values to validate.
+     * @param string $className - The name of the class to validate against.
+     *
+     * @throws ValidationListException - Throws a ValidationListException which
+     * includes a cumulative list of errors encountered during validation for each
+     * property on the object.
+     */
+    public function validate(mixed $object, string $className): void
+    {
+
+        $validators = [
+            "int"      => new IntValidator(),
+            "string"   => new StringValidator(),
+            "bool"     => new BoolValidator(),
+            "DateTime" => new DateValidator(),
+            "null"     => new NotNullValidator(),
+        ];
+
+        $arr = is_array($object) ? $object : (array) $object;
+        $validationErrors = new ValidationListException();
+
+        $ref   = new ReflectionClass($className);
+        $props = $ref->getProperties();
+
+        //For each property defined on the class.
+        foreach($props as $prop)
+        {
+            $propName    = $prop->getName();
+            $propType    = $prop->getType()->getName();
+            $propAttrs   = $prop->getAttributes();
+            $propCanNull = $prop->getType()->allowsNull();
+
+            //If the class property is passed in with the object.
+            if ($this->propIsDefined($arr, $propName))
+            {
+                //If the property is not null and a validator exists for that type make sure the type is correct.
+                $validator = array_key_exists($propType, $validators) ? $validators[$propType] : null;
+                if ($this->propIsNotNull($arr, $propName) && !is_null($validator))
+                {
+                    if (!$validator->validate($arr[$propName]))
+                    {
+                        $validationErrors->addError($validator->getErrorMessage($propName), $propName);
+                    }
+                }
+                //If the property is null make sure it is allowed to be null.
+                else if (!$propCanNull)
+                {
+                    $validationErrors->addError($validators["null"]->getErrorMessage($propName), $propName);
+                    $validationErrors->addError($validator->getErrorMessage($propName), $propName);
+                }
+
+                //If the property has any attributes make sure they are enforced.
+                foreach($propAttrs as $attr)
+                {
+                    $validatorName = $attr->getName();
+                    $attrValidator = new $validatorName(...$attr->getArguments());
```

Comment:
This should only instantiate and check `Validator` classes--classes that implement the `Validator` interface.  Otherwise the `validate` method isn't guaranteed to exist.